### PR TITLE
Include only what is really necessary in each file, add pragma once in every header, etc.

### PR DESCRIPTION
**Pull request by TuxSH**
The following pull request adds pragma once to every header as well as including only what each file needs. The changes amount to only 4 lines per file on average.

This allows developers to navigate more easily through the code, and not having to rebuild the entire library when a single file is changed.

### CODE REVIEW

File: nx/include/switch/gfx/nvgfx.h [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#pragma once
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/include/switch/gfx/nvgfx.h]: Missing types.h

File: nx/include/switch/services/fs.h [hunk 2]
```diff
@@ -1,4 +1,6 @@
 // Copyright 2017 plutoo
+#pragma once
+#include <switch/services/sm.h>
```
Comment (plutooo) [inline nx/include/switch/services/fs.h]: Missing types.h

File: nx/include/switch/services/hid.h [hunk 3]
```diff
@@ -1,5 +1,7 @@
+#pragma once
+
 #include <assert.h>
-#include <switch/types.h>
+#include <switch/services/sm.h>
```
Comment (plutooo) [inline nx/include/switch/services/hid.h]: Missing types.h

File: nx/include/switch/services/sm.h [hunk 4]
```diff
@@ -1,3 +1,7 @@
+#pragma once
```
Comment (plutooo) [inline nx/include/switch/services/sm.h]: Missing types.h

File: nx/include/switch/services/usb.h [hunk 5]
```diff
@@ -1,3 +1,6 @@
+#pragma once
+#include <switch/services/sm.h>
```
Comment (plutooo) [inline nx/include/switch/services/usb.h]: Missing types.h

File: nx/include/switch/services/vi.h [hunk 6]
```diff
@@ -1,3 +1,6 @@
+#pragma once
+#include <switch/services/sm.h>
```
Comment (plutooo) [inline nx/include/switch/services/vi.h]: Missing types.h

File: nx/source/gfx/binder.c [hunk 7]
```diff
@@ -1,5 +1,7 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/ipc.h>
```
Comment (plutooo) [inline nx/source/gfx/binder.c]: Missing types.h
Missing result.h

File: nx/source/gfx/buffer_producer.c [hunk 8]
```diff
@@ -1,5 +1,8 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/buffer_producer.c]: Missing types.h

File: nx/source/gfx/gfx.c [hunk 9]
```diff
@@ -1,5 +1,13 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/gfx.c]: Missing types.h

File: nx/source/gfx/ioctl/channel.c [hunk 10]
```diff
@@ -1,5 +1,8 @@
-#include <switch.h>
 #include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/ioctl/channel.c]: Missing types.h

File: nx/source/gfx/ioctl/nvhost-as-gpu.c [hunk 11]
```diff
@@ -1,5 +1,8 @@
-#include <switch.h>
 #include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/ioctl/nvhost-as-gpu.c]: Missing types.h

File: nx/source/gfx/ioctl/nvhost-ctrl-gpu.c [hunk 12]
```diff
@@ -1,5 +1,8 @@
-#include <switch.h>
 #include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/ioctl/nvhost-ctrl-gpu.c]: Missing types.h

File: nx/source/gfx/ioctl/nvhost-ctrl.c [hunk 13]
```diff
@@ -1,5 +1,8 @@
-#include <switch.h>
 #include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/ioctl/nvhost-ctrl.c]: Missing types.h

File: nx/source/gfx/ioctl/nvmap.c [hunk 14]
```diff
@@ -1,5 +1,8 @@
-#include <switch.h>
 #include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/gfx/ioctl/nvmap.c]: Missing types.h

File: nx/source/gfx/nvgfx.c [hunk 15]
```diff
@@ -1,6 +1,13 @@
 #include <string.h>
 #include <malloc.h>
-#include <switch.h>
+#include <switch/arm.h>
```
Comment (plutooo) [inline nx/source/gfx/nvgfx.c]: Missing types.h

File: nx/source/internal.h [hunk 16]
```diff
@@ -1,6 +1,7 @@
 #pragma once
-#include <sys/reent.h>
-#include <switch.h>
+#include <switch/arm.h>
+#include <switch/runtime/env.h>
```
Comment (plutooo) [inline nx/source/internal.h]: Why env?

Missing types.h

File: nx/source/kernel/detect.c [hunk 17]
```diff
@@ -1,5 +1,6 @@
 // Copyright 2017 plutoo
-#include <switch.h>
+#include <switch/kernel/detect.h>
```
Comment (plutooo) [inline nx/source/kernel/detect.c]: Missing types.h

File: nx/source/kernel/jit.c [hunk 18]
```diff
@@ -1,7 +1,13 @@
 // Copyright 2018 plutoo
-#include <switch.h>
 #include <malloc.h>
 
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/kernel/jit.c]: Missing types.h

File: nx/source/kernel/random.c [hunk 19]
```diff
@@ -6,7 +6,12 @@
   Modified by plutoo.
 */
 
-#include <switch.h>
+#include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/kernel/random.c]: Missing types.h

File: nx/source/kernel/shmem.c [hunk 20]
```diff
@@ -1,6 +1,9 @@
 // Copyright 2017 plutoo
-#include <switch.h>
 #include <malloc.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/kernel/shmem.c]: Missing types.h

File: nx/source/kernel/thread.c [hunk 21]
```diff
@@ -1,6 +1,10 @@
 // Copyright 2017 plutoo
-#include <switch.h>
 #include <malloc.h>
+#include <string.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/kernel/thread.c]: Missing types.h

File: nx/source/kernel/tmem.c [hunk 22]
```diff
@@ -1,6 +1,9 @@
 // Copyright 2017 plutoo
-#include <switch.h>
 #include <malloc.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/kernel/tmem.c]: Missing types.h -- uses Permission

File: nx/source/kernel/virtmem.c [hunk 23]
```diff
@@ -1,4 +1,9 @@
-#include <switch.h>
+#include <switch/result.h>
```
Comment (plutooo) [inline nx/source/kernel/virtmem.c]: Missing types.h

File: nx/source/runtime/devices/usb_comms.c [hunk 24]
```diff
@@ -1,6 +1,8 @@
 #include <string.h>
 #include <malloc.h>
-#include <switch.h>
+#include <switch/services/fatal.h>
```
Comment (plutooo) [inline nx/source/runtime/devices/usb_comms.c]: Missing types.h
Missing result.h

File: nx/source/runtime/init.c [hunk 25]
```diff
@@ -1,4 +1,10 @@
-#include <switch.h>
+#include <switch/runtime/env.h>
```
Comment (plutooo) [inline nx/source/runtime/init.c]: Missing kernel/svc.h

File: nx/source/runtime/newlib.c [hunk 26]
```diff
@@ -1,9 +1,12 @@
-#include <switch.h>
+#include <string.h>
 #include <sys/iosupport.h>
 #include <sys/time.h>
 #include <sys/lock.h>
 #include <sys/reent.h>
 #include "../internal.h"
+#include <switch/kernel/mutex.h>
```
Comment (plutooo) [inline nx/source/runtime/newlib.c]: Missing types.h

File: nx/source/services/acc.c [hunk 27]
```diff
@@ -1,5 +1,5 @@
-#include <string.h>
-#include <switch.h>
+#include <switch/services/acc.h>
```
Comment (plutooo) [inline nx/source/services/acc.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/apm.c [hunk 28]
```diff
@@ -1,5 +1,5 @@
-#include <string.h>
-#include <switch.h>
+#include <switch/services/apm.h>
```
Comment (plutooo) [inline nx/source/services/apm.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/applet.c [hunk 29]
```diff
@@ -1,5 +1,9 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/services/fatal.h>
```
Comment (plutooo) [inline nx/source/services/applet.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/bsd.c [hunk 30]
```diff
@@ -1,5 +1,7 @@
-// Copyright 2017 plutoo
-#include <switch.h>
+#include <switch/services/bsd.h>
```
Comment (plutooo) [inline nx/source/services/bsd.c]: Missing types.h
Missing result.h
Missing ipc.h

Accidentally removed Copyright notice? :-)

File: nx/source/services/fatal.c [hunk 31]
```diff
@@ -1,5 +1,7 @@
 // Copyright 2017 plutoo
-#include <switch.h>
+#include <switch/services/fatal.h>
```
Comment (plutooo) [inline nx/source/services/fatal.c]: Missing types.h
Missing result.h
Missing ipc.h
Missing kernel/svc.h

File: nx/source/services/fs.c [hunk 32]
```diff
@@ -1,5 +1,7 @@
 // Copyright 2017 plutoo
-#include <switch.h>
+#include <string.h>
+#include <switch/services/fs.h>
```
Comment (plutooo) [inline nx/source/services/fs.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/hid.c [hunk 33]
```diff
@@ -1,5 +1,9 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/services/applet.h>
```
Comment (plutooo) [inline nx/source/services/hid.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/nv.c [hunk 34]
```diff
@@ -1,5 +1,9 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/services/applet.h>
```
Comment (plutooo) [inline nx/source/services/nv.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/pm.c [hunk 35]
```diff
@@ -1,5 +1,6 @@
 // Copyright 2017 plutoo
-#include <switch.h>
+#include <switch/services/pm.h>
```
Comment (plutooo) [inline nx/source/services/pm.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/sm.c [hunk 36]
```diff
@@ -1,5 +1,6 @@
 // Copyright 2017 plutoo
-#include <switch.h>
+#include <switch/services/fatal.h>
```
Comment (plutooo) [inline nx/source/services/sm.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/usb.c [hunk 37]
```diff
@@ -1,5 +1,7 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/kernel/detect.h>
```
Comment (plutooo) [inline nx/source/services/usb.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/source/services/vi.c [hunk 38]
```diff
@@ -1,5 +1,7 @@
 #include <string.h>
-#include <switch.h>
+#include <switch/services/applet.h>
```
Comment (plutooo) [inline nx/source/services/vi.c]: Missing types.h
Missing result.h
Missing ipc.h

File: nx/include/switch/services/sm.h (hunk 4 above)
Comment (TuxSH) [reply nx/include/switch/services/sm.h]: changed

**Comment by plutooo** [comment]
Can you make it so that

#include "types.h"

instead of

#include <switch/types.h>

**Pull request merged**